resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -131,6 +131,18 @@ private[spark] class Client(
 
   require(keytab == null || !Utils.isLocalUri(keytab), "Keytab should reference a local file.")
 
+  private val ivySettings = sparkConf.getOption("spark.jars.ivySettings")
+  private val ivySettingsLocalizedFileName: Option[String] = ivySettings match {
+    case Some(ivySettingsPath) if isClusterMode && !Utils.isLocalUri(ivySettingsPath) =>
+      val ivySettingsFile = new File(ivySettingsPath)
+      require(ivySettingsFile.exists(), s"Ivy settings file $ivySettingsFile not found")
```

Comment:
+1
It would be much nicer if we could rely on `distribute` instead of trying to figure out for ourselves here if it needs to be uploaded, e.g. `distribute` will handle the local URI for us.

I think the reason we have to do this here is because `createConfArchive` doesn't give us a way to pass in the `ivySettingsLocalizedFileName` information. Can we just update `createConfArchive` to accept a parameter like `configOverrides: Map[String, String]` which provides a way to override configs that need to be different for the YARN container environment? Then we can move this within `prepareLocalResources` and rely more on the logic of `distribute`.